sklearn/cluster/spectral.py

```diff
@@ -425,12 +428,14 @@ class SpectralClustering(BaseEstimator, ClusterMixin):
       https://www1.icsi.berkeley.edu/~stellayu/publication/doc/2003kwayICCV.pdf
     """
 
-    def __init__(self, n_clusters=8, eigen_solver=None, random_state=None,
-                 n_init=10, gamma=1., affinity='rbf', n_neighbors=10,
-                 eigen_tol=0.0, assign_labels='kmeans', degree=3, coef0=1,
-                 kernel_params=None, n_jobs=None):
+    def __init__(self, n_clusters=8, eigen_solver=None, n_components=None,
+                 random_state=None, n_init=10, gamma=1., affinity='rbf',
+                 n_neighbors=10, eigen_tol=0.0, assign_labels='kmeans',
+                 degree=3, coef0=1, kernel_params=None, n_jobs=None):
         self.n_clusters = n_clusters
         self.eigen_solver = eigen_solver
+        self.n_components = n_clusters if n_components \
```

Comment:
Set `self.n_components = n_components` and pass `self.n_components` directly to `spectral_clustering` without modifying any of the parameters in `__init__`. `spectral_clustering` already handles the case when `n_components` is defined:

https://github.com/scikit-learn/scikit-learn/blob/8d3b4ff3eec890396a3d7a806bbe944f55a89cb4/sklearn/cluster/spectral.py#L256

Reply:
Nice idea! thanks for help